### ISSUE TITLE
Crash with amdgpu

### ISSUE THREAD

**Issue by greenfoo** [swaywm/sway#6290]
- **Sway Version:** 1.6
- **Debug Log:** [sway log](https://gist.github.com/greenfoo/354c32d339c7641fb7e57e73a6bcd438#file-sway-log) + [journalctl](https://gist.github.com/greenfoo/354c32d339c7641fb7e57e73a6bcd438#file-journalctl)

After a while (could be minutes, but it typically takes a few hours or more) sway becomes unresponsive. Most of these times I have to force-reset the PC, but some others I can "Ctrl+Alt+F4" into another terminal session and "kill sway" (which, by the way, is using **a lot of CPU** at that point).

This happens typically while watching YouTube videos, but doesn't need to: I have also been able to reproduce the issue scrolling through random pages from different web browsers.

Looking at the logs (more specifically, the output of "journalctl") I see there is something funny going on with the video card. It all seems to start with this message:
```
[drm:amdgpu_dm_atomic_commit_tail [amdgpu]] *ERROR* Waiting for fences timed out!
```
...which ends up forcing a GPU reset a few lines later.

The thing is that if I use Gnome I never notice this issue (maybe the GPU never suffers from this error under Gnome or maybe it does but Gnome knows how to "recover" from it without affecting the user session), but in Sway it is pretty consistent and I experience this problem at least once a day.

I did **not** suffered from this problem a few months ago but I suspect that's because there were no GPU resets back then.

Maybe there is something that can be done from sway's point of view to handle this type of GPU resets gracefully?

Thanks.

**Comment by emersion** [comment]
These are amdgpu bugs, please report to them.

**Comment by greenfoo** [comment]
Hi @emersion

While I agree that this is an amdgpu issue, please notice my last three paragraphs above: couldn't sway do something to recover from this situation (as Gnome seems to be doing) to prevent the whole session from being lost?

I mean... even after the GPU reset, if I manage to kill sway, everything is still working.

Couldn't sway try to prevent the situation where it keeps using 100% of the CPU when such a GPU reset happens?

Thanks.

**Comment by emersion** [comment]
GNOME doesn't support it, but we could try to handle GPU resets better with the robustness extension: https://github.com/swaywm/wlroots/issues/2705

**Comment by greenfoo** [comment]
That sounds great :)
Thanks.

**Comment by greenfoo** [comment]
**UPDATE**: In case someone else stumbles upon this thread because he is experiencing the same amdgpu resets, let me say that, at least in my case, the problem seem to no longer happen after **downgrading** the "linux-firmware" package to a version earlier than April (see [here](https://gitlab.freedesktop.org/drm/amd/-/issues/934) and [here](https://bbs.archlinux.org/viewtopic.php?pid=1974177#p1974177) for more details)

**Comment by kajzersoze** [comment]
In my case downgrading firmware to v20210315 didn't help.
I'm only getting these crashes in Sway, really frustrating.

**Comment by kajzersoze** [comment]
Maybe kernel version and mesa version matter as well.
But bottom line is that this happens only on Sway, shame because I had to switch to Wayfire.

**Comment by kajzersoze** [comment]
Actually, downgrading worked for me too, I made a mistake first time.
@greenfoo nice find, thanks for the info!

**Comment by TheMainMMan** [comment]
Seems like it's fixed and it doesn't happen anymore with linux-firmware-20210716

**Comment by greenfoo** [comment]
@TheMainMMan thanks for the update! I have been using `linux-firmware-20210716.b7c134f-1` for the last 24 hours since I read your message and so far so good :)

**Comment by TheMainMMan** [comment]
Had a first crash since a long time, same symptoms, there was one change though, I started to always run this PC 24/7, so this happened after more than 3 days uptime, maybe even 4, but it was 3 for sure, I'll put uptime info on Waybar, could be related, could be something else, don't really know, but it looks like that firmware problem, I've downgraded again and will see what happens next, also I will continue to run this machine 24/7

**Comment by greenfoo** [comment]
A small update: it's been one month since I upgraded (first to `linux-firmware-20210716.b7c134f-1` and later to `20210818.c46b8c3-1`, which is the version I'm currently running) and I have never experienced the hang up issue... so maybe the issue @TheMainMMan experienced was due to something else?

PS: I must say that, what I do experience (but this has **always** been the case for me, with older and newer linux-firmware versions) is that sometimes (1 out of 10?) the screen halts when waking up from a "suspend to disk". But it looks like a different issue as there is no excessive CPU usage or messages in journalctl.

**Comment by TheMainMMan** [comment]
Could be the case that it's something else, hard to tell, but I really hate when that happens so I've reverted back to the old version right away and I'm still on it, no problems at all, it's up for 7-8 days, then I update the system, reboot and repeat.
I don't really see a reason for me to upgrade linux-firmware, I might try though in next few months, but the bug is clearly there, seems like it was resolved in that latest version, it lasted way longer, but sadly, in my case, it happened again, in the exact same way. It will probably get fixed completely eventually.

I don't do suspend, so can't help you with that.
